client/src/main/java/com/vesoft/nebula/client/storage/StorageClientImpl.java

```diff
@@ -213,12 +184,12 @@ public void run() {
         return true;
     }
 
-    private boolean doPut(int space, HostAddr leader, PutRequest request) {
-        StorageService.Client client = connect(leader);
-        if (client == null) {
+    private boolean doPut(String space, HostAddr leader, PutRequest request) {
+        if (!clientMap.containsKey(addr)) {
             return false;
         }
 
+        StorageService.Client client = clientMap.get(leader);
```

Comment:
Maybe later we need to put `clientMap.get(leader);` into a separate function, it seems the client can not recover if its connection is broken (at least in jepsen test).